test/e2e/util_gatewayapi_test.go

```diff
@@ -271,6 +271,21 @@ func createGatewayClass(name, controllerName string) (*gatewayapiv1.GatewayClass
 	return gatewayClass, nil
 }
 
+// createCRD creates the CRD with the given name or retrieves it if already exists.
+func createCRD(name string) (*apiextensionsv1.CustomResourceDefinition, error) {
+	crd := buildGWAPICRDFromName(name)
+	if err := kclient.Create(context.TODO(), crd); err != nil {
+		if kerrors.IsAlreadyExists(err) {
+			if err = kclient.Get(context.TODO(), types.NamespacedName{Name: name}, crd); err == nil {
+				return crd, nil
+			}
```

Comment:
```suggestion
			if err = kclient.Get(context.TODO(), types.NamespacedName{Name: name}, crd); err != nil {
				return nil, err
			}
			return crd, nil
```
I fixed a similar logic error here: https://github.com/openshift/cluster-ingress-operator/pull/1152/commits/30b576c4293b2962b137bcb93ff77f278d7bf681

Reply:
Thanks! Fixed the logic.